core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -585,8 +585,15 @@
               "ASYMMETRIC" 'between)]
       (list 'not (list f (expr rvp-1) (expr rvp-2) (expr rvp-3))))
 
-    [:extract_expression "EXTRACT"
-     [:primary_datetime_field [:non_second_primary_datetime_field extract-field]] "FROM" ^:z es]
+    [:extract_expression "EXTRACT" [:primary_datetime_field [:non_second_primary_datetime_field extract-field]] "FROM" ^:z es]
+    ;;=>
+    (list 'extract extract-field (expr es))
+
+    [:extract_expression "EXTRACT" [:primary_datetime_field "SECOND"] "FROM" ^:z es]
+    ;;=>
+    (list 'extract "SECOND" (expr es))
+
+    [:extract_expression "EXTRACT" [:time_zone_field extract-field] "FROM" ^:z es]
```

Comment:
If you wanted to return an error in planning, you could choose to inspect the returned time object from `expr` and error based on that (Given an unambiguous mapping between SQL/XTQL/JAVA types).

Reply:
Might be worthwhile doing at _some_ point - did seem like it would be fairly complex, though? Theres quite a lot of different structures of "datetime" we might received, as I understand it (ie, not sure how to really examine say, a structure representing a function call, and figure out the return types of it, for example)

Reply:
Not sure if we have any existing examples of that in the analyzer?